src/main/java/com/launchdarkly/client/LDUser.java

```diff
@@ -70,6 +70,15 @@ JsonPrimitive getKey() {
     return key;
   }
 
+  String getKeyAsString() {
+    if (key == null || key.getAsString().equals("")) {
+      logger.warn("Found user with null/empty key: " + toString());
```

Comment:
I think this log message makes more sense in the ctor than here. Since users are immutable you'd want this logged once when a malformed user is created, not every time you access its key.